File: GreenStitch_Backend_Assignment/src/main/java/com/GreenStitch/LoginTest.java
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.By;
import org.openqa.selenium.chrome.ChromeDriver;
import org.testng.annotations.Test;

public class LoginTest {

    @Test
    public void loginTest() {
        System.setProperty("webdriver.chrome.driver", "path/to/chromedriver");
        WebDriver driver = new ChromeDriver();
        driver.get("http://example.com/login");

        WebElement username = driver.findElement(By.id("username"));
        WebElement password = driver.findElement(By.id("password"));
        WebElement loginButton = driver.findElement(By.id("loginButton"));

        username.sendKeys("testuser");
        password.sendKeys("password");
        loginButton.click();

        // Add assertions to verify login success
        driver.quit();
    }

}

File: GreenStitch_Backend_Assignment/loginTest.js
const { Builder, By, until } = require('selenium-webdriver');
const assert = require('assert');

async function testLoginFunctionality() {
  let driver = await new Builder().forBrowser('chrome').build();
  try {
    await driver.get('http://example.com/login');
    await driver.findElement(By.name('username')).sendKeys('testUser');
    await driver.findElement(By.name('password')).sendKeys('testPassword');
    await driver.findElement(By.id('loginButton')).click();
    await driver.wait(until.titleIs('Dashboard'), 10000);
    let pageTitle = await driver.getTitle();
    assert.strictEqual(pageTitle, 'Dashboard');
  } finally {
    await driver.quit();
  }
}

testLoginFunctionality();